src/core/library.c

```diff
@@ -334,6 +335,47 @@ QuicPerfCounterSnapShot(
         sizeof(PerfCounterSamples));
 }
 
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+QuicLibraryLoadRetryConfig(
+    _In_ CXPLAT_STORAGE* Storage
+    )
+{
+    //
+    // This memory layout is intentional. DO NOT REARRANGE.
+    // The Secret is immediately following the RetryConfig in memory so it
+    // can be read from the RetryConfig pointer.
+    //
+    QUIC_STATELESS_RETRY_CONFIG RetryConfig = { 0 };
+    uint8_t Secret[CXPLAT_AEAD_AES_256_GCM_SIZE] = { 0 };
+    uint32_t RotationLength = sizeof(RetryConfig.RotationMs);
+    uint32_t AlgLength = sizeof(RetryConfig.Algorithm);
+    RetryConfig.SecretLength = sizeof(Secret);
+
+    if (QUIC_SUCCEEDED(
+            CxPlatStorageReadValue(
+                Storage,
+                QUIC_SETTING_RETRY_KEY_ROTATION_MS,
+                (uint8_t*)&RetryConfig.RotationMs,
+                &RotationLength)) &&
+        QUIC_SUCCEEDED(
+            CxPlatStorageReadValue(
+                Storage,
+                QUIC_SETTING_RETRY_KEY_ALGORITHM,
+                (uint8_t*)&RetryConfig.Algorithm,
+                &AlgLength)) &&
+        QUIC_SUCCEEDED(
+            CxPlatStorageReadValue(
+                Storage,
+                QUIC_SETTING_RETRY_KEY_SECRET,
+                Secret,
+                &RetryConfig.SecretLength))) {
```

Comment:
nit: the style in this project is more like this:
```suggestion
    if (QUIC_SUCCEEDED(
        CxPlatStorageReadValue(
            Storage,
            QUIC_SETTING_RETRY_KEY_ROTATION_MS,
            (uint8_t*)&RetryConfig.RotationMs,
            &RotationLength)) &&
        QUIC_SUCCEEDED(
        CxPlatStorageReadValue(
            Storage,
            QUIC_SETTING_RETRY_KEY_ALGORITHM,
            (uint8_t*)&RetryConfig.Algorithm,
            &AlgLength)) &&
        QUIC_SUCCEEDED(
        CxPlatStorageReadValue(
            Storage,
            QUIC_SETTING_RETRY_KEY_SECRET,
            Secret,
            &RetryConfig.SecretLength))) {
```